src/hooks/useGetApiData.js

```diff
@@ -0,0 +1,15 @@
+import { useState, useEffect } from "react";
+
+const useGetApiData = (url) => {
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => setData(data));
+  }, [url]);
+
+  return [data, setData];
```

Comment:
It would be better not to return setData here. we wouldn't want the ability tp update the response data.

Reply:
resolved